src/test/java/no/ssb/crypto/tink/fpe/FpeFf3Test.java

```diff
@@ -162,4 +164,31 @@ void createKey() throws Exception {
         System.out.println(keyset);
     }
 
+    @Test
+    void ff31_encrypt_decrypt_with_different_string_encoding() throws Exception {
+        KeysetHandle keysetHandle = TinkUtil.readKeyset(KEYSET_JSON_FF31_256_ALPHANUMERIC);
+        Fpe fpe = keysetHandle.getPrimitive(Fpe.class);
+        FpeParams paramsUtf8 = FpeParams.with().unknownCharacterStrategy(SKIP);
+        FpeParams paramsLatin1 = FpeParams.with().unknownCharacterStrategy(SKIP).charset(StandardCharsets.ISO_8859_1);
+        String plaintextStr = "Lörem ïpsum dôlor sit ämêt.";  // funny characters
+
+        // utf-8
+        byte[] utf8Plaintext = plaintextStr.getBytes(StandardCharsets.UTF_8);
+        byte[] utf8Ciphertext = fpe.encrypt(utf8Plaintext, paramsUtf8);
+        byte[] utf8PlaintextRestored = fpe.decrypt(utf8Ciphertext, paramsUtf8);
+        assertThat(utf8PlaintextRestored).isEqualTo(utf8Plaintext);
+
+        // latin-1 (ISO-8859-1)
+        byte[] latin1Plaintext = plaintextStr.getBytes(StandardCharsets.ISO_8859_1);
+        byte[] latin1Ciphertext = fpe.encrypt(latin1Plaintext, paramsLatin1);
+        byte[] latin1PlaintextRestored = fpe.decrypt(latin1Ciphertext, paramsLatin1);
+        assertThat(latin1Plaintext).isEqualTo(latin1PlaintextRestored);
+
+        // Ensure the original and restored plaintexts match, regardless of the encoding used.
+        assertThat(utf8PlaintextRestored).isEqualTo(utf8PlaintextRestored);
```

Comment:
This seems to be a typo
```suggestion
        assertThat(latin1PlaintextRestored).isEqualTo(utf8PlaintextRestored);
```

Also, the comment on line 187 doesn't make sense to me, we're already checking that on line 179 and 185

Reply:
Good catch. That made no sense at all.

I have updated the test to explicitly show that the restored texts match the original plaintext, regardless of applied encoding.

(Of course one could argue that this is already shown on lines 179 and 185, and that the only diffrerence is encoding.)